apps/web/src/server/appeals/appeal-details/lpa-questionaire-review/lpa-questionnaire-review.controller.js

```diff
@@ -23,3 +28,76 @@ export async function viewLpaQuestionnaire(request, response) {
 		summaryList: { formattedSections }
 	});
 }
+
+/** @type {import('@pins/express').RequestHandler<Response>} */
+export const postLpaQuestionnaire = async (
+	{ params: { appealId, lpaQId }, body, errors },
+	response
+) => {
+	// If it fails validations
+	if (errors) {
+		const lpaQuestionnaireResponse = await getLpaQuestionnaireFromId(appealId, lpaQId);
+
+		const mappedLpaQuestionnaireSections = mapLpaQuestionnaire(lpaQuestionnaireResponse);
+		const formattedSections = [];
+		for (const section of mappedLpaQuestionnaireSections) {
+			formattedSections.push(generateSummaryList(section.header, section.rows));
+		}
+
+		return response.render('appeals/appeal/lpa-questionnaire-view.njk', {
+			summaryList: { formattedSections },
+			errors
+		});
+	}
+	// If it succeeds validation
+	try {
+		const reviewOutcome = body['review-outcome'];
+
+		// TODO: uncomment when BOAT-238 is complete, remove logger.debug if not needed
+		// await setReviewOutcomeForLpaQuestionnaire(appealId, lpaQId, reviewOutcome);
+		logger.debug(`LPA questionnaire review outcome: ${reviewOutcome}`);
+
+		response.redirect(
+			`/appeals-service/appeal-details/${appealId}/lpa-questionnaire-review/${lpaQId}/complete`
+		);
+	} catch (error) {
+		logger.error(error, 'Something went wrong when completing LPA questionnaire review');
+
+		const lpaQuestionnaireResponse = await getLpaQuestionnaireFromId(appealId, lpaQId);
+
+		const mappedLpaQuestionnaireSections = mapLpaQuestionnaire(lpaQuestionnaireResponse);
+		const formattedSections = [];
+		for (const section of mappedLpaQuestionnaireSections) {
+			formattedSections.push(generateSummaryList(section.header, section.rows));
+		}
+
+		return response.render('appeals/appeal/lpa-questionnaire-view.njk', {
+			summaryList: { formattedSections },
+			error
+		});
+	}
+};
+
+/**
+ *
+ * @param {*} request
+ * @param {*} response
+ */
+export async function getLpaQuestionnaireReviewComplete(
+	{ params: { appealId, lpaQId } },
+	response
+) {
+	/*
+	TODO:
+	- uncomment when generic "complete" page is done
+	- provide correct path to its .njk
+	- remove logger.debug
+	- write test for the review complete page
+	*/
+	logger.debug(`LPA questionnaire review complete page: ${appealId}, ${lpaQId}`);
+
+	response.render('appeals/appeal/generic-complete-page.njk', {
```

Comment:
Does this throw an error currently? The referenced page template doesn't exist, which is clearly intentional, but just wondering what happens when this render is called?

Reply:
Yes, it does throw error now